wsgi.py

```diff
@@ -6,255 +6,348 @@
 import re
 import os
 from flask import Flask, redirect, url_for, request, Response
-#from flask_restful import reqparse
+from flask_httpauth import HTTPBasicAuth
+
+# from flask_restful import reqparse
 
 import sys
 
-from openshift_rolebindings import *
-from openshift_project import *
-from openshift_identity import *
-from openshift_user import *
+from openshift import *
 
 application = Flask(__name__)
+auth = HTTPBasicAuth()
 
-if __name__ != '__main__':
-    gunicorn_logger = logging.getLogger('gunicorn.error')
+if __name__ != "__main__":
+    gunicorn_logger = logging.getLogger("gunicorn.error")
     application.logger.handlers = gunicorn_logger.handlers
     application.logger.setLevel(gunicorn_logger.level)
 
 
-def get_user_token():
-    with open('/var/run/secrets/kubernetes.io/serviceaccount/token', 'r') as file:
+def get_openshift():
+    version = os.environ["OPENSHIFT_VERSION"]
+    url = os.environ["OPENSHIFT_URL"]
+    if version == 3:
+        shift = openshift_3_x(url, version, application.logger)
+        application.logger.info("using Openshift ver 3")
+    else:
+        shift = openshift_4_x(url, version, application.logger)
+        application.logger.info("using Openshift ver 4")
+    with open("/var/run/secrets/kubernetes.io/serviceaccount/token", "r") as file:
         token = file.read()
-        return token
-    return ""
-
-def get_token_and_url():
-    token = get_user_token()
-    openshift_url = os.environ["openshift_url"]
-    return (token, openshift_url)
-
-@application.route("/users/<user_name>/projects/<project_name>/roles/<role>", methods=['GET'])
+        shift.set_token(token)
+        application.logger.info("Attached token to shift")
+        return shift
+    return None
```

Comment:
Not really.  Return None seems to be the default behavior of Python, though I am just being explicit in the behavior that I want.

Reply:
I can't see how this code will reach the `return None` statement.

>  If there is no token associated with the shift object, I don't want it to be used

Where are you checking that?

Reply:
I will add the check that you requested, though it is unnecessary.

the object shift, however is not to be returned if there is no token assigned to it.  That is by design, ultimately, it means that the project was mis-installed.

Reply:
Ok, here is the full context of the function that you are referring to

`def get_openshift():
    version = os.environ["OPENSHIFT_VERSION"]
    url = os.environ["OPENSHIFT_URL"]
    if version == 3:
        shift = openshift_3_x(url, version, application.logger)
        application.logger.info("using Openshift ver 3")
    else:
        shift = openshift_4_x(url, version, application.logger)
        application.logger.info("using Openshift ver 4")
    with open("/var/run/secrets/kubernetes.io/serviceaccount/token", "r") as file:
        token = file.read()
        shift.set_token(token)
        application.logger.info("Attached token to shift")
        return shift
`
what happens when the open fails to open the file?  This token is not in the container, it is in a secret that is mounted for reading in the container.  This is the token that gives the code access to the the service account.  If that token doesn't exist, I do not want the shift object passed back as it is only partially initialized.  Perhaps what you really wanted to say is not split the initialization into 2 parts, as in the following:

`def get_openshift():
    version = os.environ["OPENSHIFT_VERSION"]
    url = os.environ["OPENSHIFT_URL"]
    with open("/var/run/secrets/kubernetes.io/serviceaccount/token", "r") as file:
        token = file.read()
        if version == 3:
            shift = openshift_3_x(token, url, version, application.logger)
            application.logger.info("using Openshift ver 3")
        else:
            shift = openshift_4_x(token, url, version, application.logger)
            application.logger.info("using Openshift ver 4")
        return shift
    application.logger.info.error("Unable to open service account token file, shift initialization incomplete")
`

But here, again, this function has an implied return None at the end, it is just not explicitly stated.

Reply:
Well, pylon gives me the following informational:

wsgi.py:37:0: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements).

Reply:
Actually, I'm fine with the exception as that will cause a failure when the project is misconfigured.

If it is misconfigured it will not run as expected.